src/unix/linux-core.c

```diff
@@ -1015,14 +1016,261 @@ uint64_t uv_get_total_memory(void) {
 }
 
 
-static uint64_t uv__read_cgroups_uint64(const char* cgroup, const char* param) {
-  char filename[256];
+#define PROC_SELF_MOUNTINFO "/proc/self/mountinfo"
+#define PROC_SELF_CGROUP "/proc/self/cgroup"
+#define CGROUPS_VERSION_UNKNOWN 0x0
+#define CGROUPS_VERSION_1 0x1
+#define CGROUPS_VERSION_2 0x2
+#define CGROUPS_V2_CTRL_PREFIX "0::"
+#define CGROUPS_V2_CTRL_PREFIX_LEN 3
+#define CGROUPS_V2_NO_LIMIT_VALUE "max"
+#define CGROUPS_BUF_SIZE 4096
+
+
+typedef struct {
+  /* cgroups version, one of CGROUPS_VERSION_*. */
+  uint8_t cgroups_version;
+  /* Path in which param files are found for a subsystem. */
+  char path[CGROUPS_BUF_SIZE];
+} uv__cgroups_subsystem_info_t;
+
+
+/*
+ * Writes `start` to `field_start` and the length of the current
+ * whitespace-delimited field to `field_len`, and returns an pointer to the
+ * remainder of the string (excluding the whitespace).
+ * Note that we assume that `start` ends with "\n\0".
+ */
+static const char* uv__mountinfo_next_field(const char* start,
```

Comment:
I would suggest replacing this function with `strsep`. Since that is not portable, you can implement your own version, or use a public domain copy.  It would be better to have a copy of `strsep` rathan than a function that is functionally equivalent but subtly different.